src/utils/timer.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  addTimedelta,
+  TimeComponents,
+  toMilliseconds
+} from "./time"
+
+export type TimerComponentsParameters = {
+  duration: TimeComponents,
+  tick?: TimeComponents,
+  continues?: boolean,
+  onTick?: (timer: Timer) => void,
```

Comment:
Should we also offer an await-able interface? E.g.
```typescript
nextTick?: Promise<Timer>
nextComplete: Promise<Timer>
nextStart?: Promise<Timer>
```

Reply:
No, this is not a promise but a callback. A promise cannot be triggered twice but can be rejected. A callback not.

Reply:
Yes, I know. But should we _also_? That is: in addition to the existing functions?

(If we'd be using RxJs I'd ask for an `Observable` interface instead, because that one already resolves n times before completing, but then I'd just use [Observable.interval](https://rxjs.dev/api/index/function/interval) and probably not use this timer.)

Callbacks are not chainable. Neither condition (being non-chainable and triggering only once) is a dealbreaker, of course, but it might be sensible to let the user choose their poison.
(_Might_, that is. Feel free to dismiss. Just figured, that it would be nice: I myself would probably choose await-and-resubscribe over callback, when already inside an await-heavy callstack, but would choose callback in code that is not yet `async`. That said, I could just add this interface myself later on.)

Reply:
Callbacks are the minimal approach and the best fitting solution here. Proof me wrong when Chaining, Promising or Oberservers are nice patterns but YAGNI.
There is maybe away to get rid of the `onTick` callback. Let me check that out.

Reply:
Fair enough.